### PR TITLE
Feature/idva5 1339 adapt check your answer page id docs

### PR DESCRIPTION
**Pull request by cbroadley-ch**
__Short description outlining key changes/additions__

Updating check your answers page to dynamically display details of the Identity documents selected in Group A and B.
This involves the addition of 1 method within the `formatService.ts` to store and format an array of Group B identity documents. 
These methods are implemented within the` checkYourAnswers.ts` controller to display the relevant Identity Document details on the `check-your-answers.njk` front end view

**Update:**

The `DocumentDetails.ts` model has been updated to include an extra parameter for formattedExpiryDate, this is used to display the expiry date previously entered on the ID Document Details page in a human readable format when they reach the Check your answers page.

**Group A:** 

When a user selects Identity Documents from Group A (either through Option 1 or Option 2 route) and inputs a valid  Document Number / Expiry Date / Country of Issue then this will display these details within an additional `govukSummaryList` towards the bottom of the front end UI on the `check-your-answers.njk` front end view.

Each Group A document displayed on the Check Your Answers page includes a 'Change' link, allowing the user to navigate back to the Identity Details input page and update these accordingly. Please note, this will navigate the user directly to the document details they wish update upon clicking the relevant 'Change' link (i.e. If the user selects the 'Change' link beside the Passport 'Country of Issue' field).

**Group B:**

If the user has previously selected Option 2 and selected Identity documents from 'Group B' then a new row is generated within the `check-your-answers.njk` front end view (between _How the identity documents were checked_ row and _Identity verification declaration row_ ). This row displays the names of the documents they have selected from Group B.

If no Group B documents were selected, or the user has deselected Group B documents (after clicking the 'Change' link on the relevant row within Check your answers page) then this row is no longer visible to the user)

__JIRA Ticket Number__

[IDVA5-1339](https://companieshouse.atlassian.net/browse/IDVA5-1339?atlOrigin=eyJpIjoiNjczMTBiNThlYjA0NDgyNGE2NzU5NDdiMDdkN2VkYWYiLCJwIjoiaiJ9)

### Type of change

* [ ] Bug fix
* [x] New feature
* [ ] Breaking change
* [ ] Infrastructure change

### Pull request checklist

* [x] I have added unit tests for new code that I have added
* [x] I have added/updated functional tests where appropriate
* [x] The code follows our [coding standards](https://github.com/companieshouse/styleguides/blob/master/java.md)

__An exhaustive list of peer review checks can be read [here](https://github.com/companieshouse/styleguides/blob/master/java_review.md#developer-actions-prior-to-code-commitreview-started)__

[IDVA5-1339]: https://companieshouse.atlassian.net/browse/IDVA5-1339?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by ch-code-analysis** [comment]
CI: No security warnings found

**Comment by ch-code-analysis** [comment]
CI: No security warnings found

**Comment by ch-code-analysis** [comment]
CI: No security warnings found

File: src/services/formatService.ts [hunk 1]
```diff
@@ -56,6 +57,24 @@ export class FormatService {
         return new Intl.DateTimeFormat("en-GB", options).format(date);
     }
 
+    public static formatDateIdentityDocuments (documents?: DocumentDetails[]) {
+        const formattedDocument: any[] = [];
```
Comment (sampankumar) [inline src/services/formatService.ts:61]: instead of any type can formattedDocument be of DocumentDetails type?

**Comment by ch-code-analysis** [comment]
CI: No security warnings found

File: src/services/formatService.ts (hunk 1 above)
Comment (cbroadley-ch) [reply src/services/formatService.ts]: This method has now been removed, thanks. 
I have followed your suggestion here and included an extra field `'formattedExpiryDate'` within the DocumentDetails object that takes in a String and is used to display the expiry date in a readable format when on the Check your answers page

**Comment by ch-code-analysis** [comment]
CI: No security warnings found

**Review by sampankumar — APPROVED** [review]

**Review by sardhani-ch — APPROVED** [review]

**Pull request merged**